Orange/widgets/data/owrandomize.py

```diff
@@ -0,0 +1,119 @@
+import random
+
+from AnyQt.QtCore import Qt
+from AnyQt.QtWidgets import QSizePolicy
+
+from Orange.data import Table
+from Orange.preprocess import Randomize
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget
+from Orange.widgets import gui
+
+
+class OWRandomize(OWWidget):
+    name = "Randomize"
+    description = "Randomize features, class and/or metas in data table."
+    icon = "icons/Random.svg"
+    priority = 2100
+
+    inputs = [("Data", Table, "set_data")]
+    outputs = [("Data", Table)]
+
+    resizing_enabled = False
+    want_main_area = False
+
+    shuffle_class = Setting(1)
+    shuffle_attrs = Setting(0)
+    shuffle_metas = Setting(0)
+    scope_prop = Setting(80)
+    random_seed = Setting(0)
+    auto_apply = Setting(True)
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+
+        # GUI
+        box = gui.vBox(self.controlArea, "Shuffle")
+        self.class_check = gui.checkBox(
+            box, self, "shuffle_class", "Classes",
+            callback=self._shuffle_check_changed)
+        self.attrs_check = gui.checkBox(
+            box, self, "shuffle_attrs", "Features",
+            callback=self._shuffle_check_changed)
+        self.metas_check = gui.checkBox(
+            box, self, "shuffle_metas", "Metas",
+            callback=self._shuffle_check_changed)
+        gui.separator(box, 10, 10)
+        self.replicable_check = gui.checkBox(
+            box, self, "random_seed", "Replicable shuffling",
+            callback=self._shuffle_check_changed)
+
+        box = gui.vBox(self.controlArea, "Scope")
+        hbox = gui.hBox(box)
+        gui.widgetLabel(hbox, "No data")
+        self.scope_slider = gui.hSlider(
+            hbox, self, "scope_prop", minValue=0, maxValue=100, width=140,
+            createLabel=False, callback=self._scope_slider_changed)
+        gui.widgetLabel(hbox, "All data")
+        self.scope_label = gui.widgetLabel(
+            box, "", alignment=Qt.AlignCenter,
+            sizePolicy=(QSizePolicy.MinimumExpanding, QSizePolicy.Fixed))
+        self._set_scope_label()
+
+        self.apply_button = gui.auto_commit(
+            self.controlArea, self, "auto_apply", "&Apply",
+            box=False, commit=self.apply)
+
+    def _shuffle_check_changed(self):
+        self.apply()
+
+    def _scope_slider_changed(self):
+        self._set_scope_label()
+        self.apply()
+
+    def _set_scope_label(self):
+        self.scope_label.setText("{}%".format(self.scope_prop))
+
+    def set_data(self, data):
+        self.data = data
+        self.apply()
+
+    def apply(self):
+        data = None
+        if self.data:
+            rand_seed = self.random_seed or None
+            _type = Randomize.RandomizeClasses * self.shuffle_class | \
+                    Randomize.RandomizeAttributes * self.shuffle_attrs | \
+                    Randomize.RandomizeMetas * self.shuffle_metas
```

Comment:
This is exactly how I'd do it and I like it. But perhaps

    type_ = sum(part for part, flag in zip(Randomize.Type, [self.shuffle_class, shuffle_attrs, shuffle_metas]) if flag)

is better. (If you use `[self.shuffle_class, shuffle_attrs, shuffle_metas]` in a lot of places, you can create a property

    @property
     def parts(self):
         return [self.shuffle_class, shuffle_attrs, shuffle_metas]

and simplify this expression and also `send_report`.